### PR TITLE
angie: add withAcme option

### PR DESCRIPTION
**Pull request by Izorkin**
## Description of changes
Add withAcme option.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Aleksanaa** [comment]
@ofborg build angie

File: pkgs/servers/http/angie/default.nix [hunk 1]
```diff
@@ -3,21 +3,25 @@
...
     url = "https://download.angie.software/files/angie-${version}.tar.gz";
-    hash = "sha256-g6PyuyulnltnZJWiZ01iYG1k6Lz5nO+gneb8M4q3WHo=";
+    hash = "sha256-pSBy+Gv3rSa0WqQOxLHM0V03/vLqvOYBGxrjY9qsBfQ=";
   };
 
-  configureFlags = lib.optional withQuic [
+  configureFlags = lib.optionals withAcme [
+    "--with-http_acme_module"
+    "--http-acme-client-path=/var/lib/nginx/acme"
+  ] ++ lib.optional withQuic [
```
Comment (drupol) [inline pkgs/servers/http/angie/default.nix:24]:
```suggestion
  ] ++ lib.optionals withQuic [
```

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/servers/http/angie/default.nix (hunk 1 above)
Comment (Izorkin) [reply pkgs/servers/http/angie/default.nix:24]: Isn't it recommended to specify `lib.optional` when there is one parameter, and `lib.optionals` when there are several?

Comment (drupol) [reply pkgs/servers/http/angie/default.nix:24]: Not that I have seen so far or know of.

In your changes, you're wrapping twice the value in a list, that's precisely why I request a change.

I wouldn't recommend using lib.optional here.

Comment (Aleksanaa) [reply pkgs/servers/http/angie/default.nix:24]:
```
nix-repl> :p lib.optional true [ "something" ]
[
  [ "something" ]
]

nix-repl> :p lib.optionals true [ "something" ]
[ "something" ]
```

Comment (Izorkin) [reply pkgs/servers/http/angie/default.nix]: Update.

**Review by drupol — APPROVED** [review]

File: pkgs/servers/http/angie/default.nix (hunk 1 above)
Comment (drupol) [reply pkgs/servers/http/angie/default.nix]: Thank you

Comment (Izorkin) [reply pkgs/servers/http/angie/default.nix]: In Nginx uses `lib.optional` and `lib.optionals` too:
```nix
  ] ++ lib.optionals withPerl [
    "--with-http_perl_module"
    "--with-perl=${perl}/bin/perl"
    "--with-perl_modules_path=lib/perl5"
  ] ++ lib.optional withSlice "--with-http_slice_module"
    ++ lib.optional (gd != null) "--with-http_image_filter_module"
    ++ lib.optional (geoip != null) "--with-http_geoip_module"
    ++ lib.optional (withStream && geoip != null) "--with-stream_geoip_module"
    ++ lib.optional (with stdenv.hostPlatform; isLinux || isFreeBSD) "--with-file-aio"
    ++ configureFlags
    ++ map (mod: "--add-module=${mod.src}") modules;

```

Comment (drupol) [reply pkgs/servers/http/angie/default.nix]: That's something I personally don't like but I don't see any double wrapping in there, so I wouldn't request a chance.

**Pull request merged**